tests/conftest.py

```diff
@@ -1,31 +1,109 @@
-import pytest 
+import pytest
 import os
-import pandas as pd
-from datetime import datetime
-from numpy import ndarray
+from flask_sqlalchemy import SQLAlchemy
+from typing import List
 
 from controllers import app as client
 from controllers import db as database
-from models import Data, QAModel
+
+from models import *
+from services import *
+from dtos import *
+
+from services.visitor_service import VisitorService
+from tests.mock.dtos.v2 import MockDataCreateRequestV2
+from tests.mock.dtos.v2 import MockQAModelCreateRequestV2
+from tests.mock.dtos.v2.mock_visitor_dto_v2 import MockVisitorCreateRequestV2
 
 tester = client.app.test_client()
 
+
 @pytest.fixture
 def db():
-    """
-    Fixture to yield use of the database
-    """
-
     yield database
 
+
 @pytest.fixture
 def app():
-    """
-    Fixture to yield use of the tester 
-    """
-
     yield tester
 
+
+@pytest.fixture
+def data_model(db: SQLAlchemy):
+    dto: DataCreateRequestV2 = MockDataCreateRequestV2()
+    data: Data = dto.to_model()
+
+    yield data
+
+    db.session.delete(data)
+    db.session.commit()
+
+
+@pytest.fixture
+def data_model_list(db: SQLAlchemy):
+    data_list: List[Data] = [MockDataCreateRequestV2().to_model()
+                             for _ in range(10)]
+
+    yield data_list
+
+    for data in data_list:
+        db.session.delete(data)
+        db.session.commit()
+
+
+@pytest.fixture
+def data_service():
+    service = DataService()
+    yield service
+
+
+@pytest.fixture
+def qa_model_model(db: SQLAlchemy):
+
+    dto: QAModelCreateRequestV2 = MockQAModelCreateRequestV2()
+    qa_model = dto.to_model()
```

Comment:
Should probably just do this in one line, qa_model: QAModel = MockQAModelCreateRequestV2().to_model()